test/Ocelot.Benchmarks/LoggingBenchmarks.cs

```diff
@@ -0,0 +1,256 @@
+                logging.SetMinimumLevel(minLogLevel);
+                logging.AddSerilog(_logger);
+            })
+            .Configure(app =>
+            {
+                app.Use(async (context, next) =>
+                {
+                    var loggerFactory = context.RequestServices.GetService<IOcelotLoggerFactory>();
+                    var ocelotLogger = loggerFactory.CreateLogger<LoggingBenchmarks>();
+                    ocelotLogger.LogDebug(() => $"DEBUG: {nameof(ocelotLogger)},  {nameof(loggerFactory)}");
+                    ocelotLogger.LogTrace(() => $"TRACE: {nameof(ocelotLogger)},  {nameof(loggerFactory)}");
+                    ocelotLogger.LogInformation(() => $"INFORMATION: {nameof(ocelotLogger)},  {nameof(loggerFactory)}");
+                    ocelotLogger.LogWarning(() => $"WARNING: {nameof(ocelotLogger)},  {nameof(loggerFactory)}");
+                    ocelotLogger.LogError(() => $"ERROR: {nameof(ocelotLogger)},  {nameof(loggerFactory)}",
+                        new Exception("test"));
+                    ocelotLogger.LogCritical(() => $"CRITICAL: {nameof(ocelotLogger)},  {nameof(loggerFactory)}",
+                        new Exception("test"));
+
+                    await next.Invoke();
+                });
+                app.UseOcelot().Wait();
+            })
+            .Build();
+
+        _webHost.Start();
+    }
+
+    private void OcelotFactory(LogLevel minLogLevel)
+    {
+        var configuration = new FileConfiguration
+        {
+            Routes = new List<FileRoute>
+            {
+                new()
+                {
+                    DownstreamPathTemplate = "/",
+                    DownstreamHostAndPorts = new List<FileHostAndPort>
+                    {
+                        new()
+                        {
+                            Host = "localhost",
+                            Port = 51879,
+                        },
+                    },
+                    DownstreamScheme = "http",
+                    UpstreamPathTemplate = "/",
+                    UpstreamHttpMethod = new List<string> { "Get" },
+                },
+            },
+        };
+
+        GivenThereIsAConfiguration(configuration);
+        GivenThereIsAServiceRunningOn("http://localhost:51879", "/", 201, string.Empty);
+        GivenOcelotIsRunning("http://localhost:5000", minLogLevel);
+    }
+
+    public static void GivenThereIsAConfiguration(FileConfiguration fileConfiguration)
+    {
+        var configurationPath = Path.Combine(AppContext.BaseDirectory, "ocelot.json");
+
+        var jsonConfiguration = JsonConvert.SerializeObject(fileConfiguration);
+
+        if (File.Exists(configurationPath))
+        {
+            File.Delete(configurationPath);
+        }
+
+        File.WriteAllText(configurationPath, jsonConfiguration);
+    }
+
+    private void GivenThereIsAServiceRunningOn(string baseUrl, string basePath, int statusCode, string responseBody)
```

Comment:
My search results showed me 9 definitions in the solution!
Why not to define public shared helper in `Ocelot.Testing` project to reuse in both Acceptance & Benchmarks projects?

I mean, if it is possible (the same bodies) then why not to move the definition to common testing project?

Reply:
Yeah, we should open a new issue and refactor it. For this PR, it's a bit too late and too dangerous